src/ECER.Managers.Registry/ApplicationHandlers.cs

```diff
@@ -197,4 +199,12 @@ public async Task<ReferenceSubmissionResult> Handle(Contract.Applications.OptOut
     ecerContext.CommitTransaction();
     return ReferenceSubmissionResult.Success();
   }
+
+  public async Task<ApplicationStatusQueryResults> Handle(ApplicationStatusQuery request, CancellationToken cancellationToken)
+  {
+    ArgumentNullException.ThrowIfNull(request);
+
+    var applicationStatus = await applicationRepository.Status(new Resources.Documents.Applications.ApplicationStatusQuery(request.Id), cancellationToken);
```

Comment:
You should already have a query for an application, the manager handler is exactly the place to project only the status if needed